esp-hal/src/lcd_cam/lcd/mod.rs

```diff
@@ -4,13 +4,566 @@
+                    .lcd_clkm_div_b().variant(div_b as _)
+            );
+    }
+
+    /// Configure the PCLK clock level that should be used while no transaction
+    /// is active.
+    ///
+    /// # Arguments
+    /// * `level` - PCLK clock level during idle, where high(`true`) and
+    /// low(`false`).
+    #[inline(always)]
+    pub fn set_clock_idle_level(&self, level: bool) {
+        self.lcd_cam
+            .lcd_clock()
+            .modify(|_, w| w.lcd_ck_idle_edge().bit(level));
+    }
+
+    /// Set the PCLK sample edge
+    ///
+    /// # Arguments
+    /// * `phase` - Edge phase of the pixel clock at which to sample
+    #[inline(always)]
+    pub fn set_pixel_clock_edge(&self, phase: Phase) {
+        self.lcd_cam
+            .lcd_clock()
+            .modify(|_, w| w.lcd_ck_out_edge().bit(phase == Phase::ShiftLow));
+    }
+
+    /// Set pixel clock prescaler.
+    /// This configures the ratio between the configured LCD peripheral clock
+    /// and the pixel clock, which is used to push out pixels with.
+    /// Essentially, this is a second clock divider: PCLK = LCD_CLK /
+    /// prescale
+    #[inline(always)]
+    pub fn set_pixel_clock_prescale(&self, prescale: u32) {
+        debug_assert!(prescale > 0);
+        debug_assert!(prescale <= LCD_PCLK_DIV_MAX);
+        // Formula: pixel_clk = lcd_clk / (1 + clkcnt_n)
+        // clkcnt_n can't be zero
+        let clkcnt_n = (prescale - 1).max(0);
+        // signals that pclk == lcd_clk
+        let lcd_clk_equ_sysclk = prescale == 1;
+        self.lcd_cam.lcd_clock().modify(|_, w| {
+            w.lcd_clk_equ_sysclk()
+                .bit(lcd_clk_equ_sysclk)
+                .lcd_clkcnt_n()
+                .variant(clkcnt_n as _)
+        });
+    }
+
+    /// Enable/Disable the LCD peripheral's internal RGB-YUV color format
+    /// converter.
+    pub fn enable_rgb_yuv_converter(&self, enabled: bool) {
+        self.lcd_cam
+            .lcd_rgb_yuv()
+            .modify(|_, w| w.lcd_conv_bypass().bit(enabled));
+    }
+
+    // TODO: rest of the yuv-rgb convert stuff
+
+    /// Configure the clock cycles to spend in each of the three transaction
+    /// phases. For RGB-Mode, all phases except the data phase are disabled.
+    ///
+    /// # Arguments
+    /// * `cmd_cycles` - Clock cycles to spend in the CMD phase
+    /// * `dummy_cycles` - Clock cycles to spend in the CMD phase
+    /// * `data_cycles` - Clock cycles to spend in the DATA phase.
+    #[inline(always)]
+    pub fn set_phase_cycles(
+        &self,
+        cmd_cycles: Option<u32>,
+        dummy_cycles: Option<u32>,
+        data_cycles: Option<u32>,
+    ) {
+        self.lcd_cam.lcd_user().modify(|_, w| {
+            if let Some(cmd_cycles) = cmd_cycles {
+                debug_assert!(cmd_cycles <= 2);
+                w.lcd_cmd()
+                    .bit(cmd_cycles > 0)
+                    .lcd_cmd_2_cycle_en()
+                    .bit(cmd_cycles > 1);
+            }
```

Comment:
Don't you need an else here?

Reply:
The idea behind this interface with the optionals was basically that you can decide on calling-point which of the parameters you want to change. So basically:

```rust
set_phase_cycles(Some(0), None, Some(1));
```

means: Set `cmd_cycles` to `0`, set `data_cycles` to `1`, and leave `dummy_cycles´ unchanged.

Reply:
`set` is the wrong verb, `modify`, `adjust` or `update` would be better.